internal/controller/serverclaim_controller.go

```diff
@@ -514,16 +514,23 @@ func (r *ServerClaimReconciler) enqueueServerClaimByRefs() handler.EventHandler
 			return nil
 		}
 		for _, claim := range claimList.Items {
-			if claim.Spec.ServerRef != nil && claim.Spec.ServerRef.Name == host.Name {
+			if claim.Spec.ServerRef != nil && claim.Spec.ServerRef.Name == server.Name {
 				req = append(req, reconcile.Request{
 					NamespacedName: types.NamespacedName{Namespace: claim.Namespace, Name: claim.Name},
 				})
 				return req
 			}
-			if claim.Spec.ServerRef == nil {
-				req = append(req, reconcile.Request{
-					NamespacedName: types.NamespacedName{Namespace: claim.Namespace, Name: claim.Name},
-				})
+			if claim.Spec.ServerRef == nil && claim.Spec.ServerSelector != nil {
```

Comment:
Checking for `claim.Spec.ServerSelector != nil` causes best-effort claims to not be queued. Is that intended?

Reply:
Fair point. Reverted that part.